src/main/java/bio/terra/service/auth/iam/IamResourceType.java

```diff
@@ -28,41 +32,37 @@ public String toString() {
     return samResourceName;
   }
 
+  @FunctionalInterface
+  private interface FilterFunction {
+    boolean apply(IamResourceType resourceType);
+  }
+
+  private static IamResourceType findIamResourceType(FilterFunction filter, String typeName) {
+    return Arrays.stream(IamResourceType.values())
+        .filter(filter::apply)
+        .findFirst()
+        .orElseThrow(() -> new RuntimeException("Invalid resource type: " + typeName));
```

Comment:
This refactor looks good, although I would preserve the existing behavior of `fromValue()` where `null` is returned on failure.

Reply:
Another approach would be to return `null` and then callers could write
```
  Objects.requireNonNull(findIamResourceType(...), () -> "Invalid resource type: " + typeName);
```